addons/stock/models/stock_picking.py

```diff
@@ -738,7 +738,9 @@ def action_assign(self):
         @return: True
         """
         self.filtered(lambda picking: picking.state == 'draft').action_confirm()
-        moves = self.mapped('move_lines').filtered(lambda move: move.state not in ('draft', 'cancel', 'done'))
+        moves = self.mapped('move_lines').filtered(lambda move: move.state not in ('draft', 'cancel', 'done')).sorted(
+            key=lambda ml: (-int(ml.priority), not bool(ml.date_deadline), ml.date_deadline, ml.id)
```

Comment:
Ok just keep the lambda move as upper since we use ml often for the object `stock.move.line` it's distrurbing

Reply:
Indeed, we should maybe create a feedback to rename the variable too :D

Reply:
It's already done in master

Reply:
:ok_hand: (and it's done)